### ISSUE TITLE
Investigate the trello clones in github

### ISSUE THREAD

**Issue by guyboertje** [guyboertje/kanghi#1]
- https://github.com/somlor/kanban

**Comment by guyboertje** [comment]
They all seem to be json api backbone single page client side apps
Is this a logical way forward?
Will make a ios/android app easier.
Backbone, Ember, Spine, ReactJs, Angular
So much variety, whats the new hotness?

**Comment by guyboertje** [comment]
Downsides:
- Javascript testing
- Javascript (use Opal?)
- Coffescript in Lotus?
- Rest API in Lotus?

**Comment by guyboertje** [comment]
ReactJs is a strong contender...
http://www.crashlytics.com/blog/building-user-interfaces-with-react/
goes into problems with clientside apps and particularly UI state machines

**Comment by guyboertje** [comment]
Maybe, if we go down the Server is a JSON Rest API endpoint, consider Roda - http://roda.jeremyevans.net/index.html

**Comment by guyboertje** [comment]
http://en.wikipedia.org/wiki/Nested_set_model
http://explainextended.com/2009/09/24/adjacency-list-vs-nested-sets-postgresql/

**Comment by guyboertje** [comment]
OTOH, Rails will have the better ReactJs toolchain and, with Grape, will have a trivial API setup.

So too for Neo4j graph db support.

**Comment by guyboertje** [comment]
https://hackhands.com/react-rails-tutorial/

**Comment by rob-murray** [comment]
:+1: For Rails & React

**Comment by guyboertje** [comment]
Yeah, it looks like Lotus does not yet have a working asset pipeline.
Ditto for Roda.

+1 for Rails n React.

@swathik, @dimitrismistriotis: votes?

**Comment by guyboertje** [comment]
http://fluxxor.com/ to manage the flow of data from the API call to the React components.
Any other candidates?

**Comment by guyboertje** [comment]
http://webpack.github.io/ as mentioned in the Rails + React tutorial.

**Comment by swathik** [comment]
Okay with Rails and React.

**Comment by guyboertje** [comment]
Graph db neo4j

http://helabs.com/blog/2014/07/25/starting-with-graph-databases-using-neo4j-and-rails/

**Comment by guyboertje** [comment]
its done, switched to rails and react, added webpack

**Comment by guyboertje** [comment]
http://www.railsonmaui.com/blog/2014/10/02/integrating-webpack-and-the-es6-transpiler-into-an-existing-rails-project/

**Comment by guyboertje** [comment]
http://react-bootstrap.github.io/getting-started.html
